Kernel/Locking/Spinlock.h

```diff
@@ -13,18 +13,34 @@
 
 namespace Kernel {
 
+template<LockRank Rank>
```

Comment:
Same here, instead of moving everything to the header.

Reply:
Whenever we add a lock rank (we will probably need more than what we have right now), we'd need to add another instantiation so I'm not sure what benefit that provides.

Reply:
That is indeed true. Consider this comment and the other one about template instantiation as dismissed then. :^)